pkg/reconciler/broker/broker.go

```diff
@@ -0,0 +1,480 @@
+		return err
+	}
+	if exists {
+		if err := topic.Delete(ctx); err != nil {
+			logger.Error("Failed to delete Pub/Sub topic", zap.Error(err))
+			return err
+		}
+		logger.Info("Deleted PubSub topic", zap.String("name", topic.ID()))
+		r.Recorder.Eventf(b, corev1.EventTypeNormal, topicDeleted, "Deleted PubSub topic %q", topic.ID())
+	}
+
+	// Delete pull subscription if it exists.
+	// TODO could alternately set expiration policy to make pubsub delete it after some idle time.
+	// https://cloud.google.com/pubsub/docs/admin#deleting_a_topic
+	subID := resources.GenerateDecouplingSubscriptionName(b)
+	sub := client.Subscription(subID)
+	exists, err = sub.Exists(ctx)
+	if err != nil {
+		logger.Error("Failed to verify Pub/Sub subscription exists", zap.Error(err))
+		return err
+	}
+	if exists {
+		if err := sub.Delete(ctx); err != nil {
+			logger.Error("Failed to delete Pub/Sub subscription", zap.Error(err))
+			return err
+		}
+		logger.Info("Deleted PubSub subscription", zap.String("name", sub.ID()))
+		r.Recorder.Eventf(b, corev1.EventTypeNormal, subDeleted, "Deleted PubSub subscription %q", sub.ID())
+	}
+
+	return nil
+}
+
+// reconcileTriggers reconciles the Triggers that are pointed to this broker
+func (r *Reconciler) reconcileTriggers(ctx context.Context, b *brokerv1beta1.Broker) error {
+
+	// TODO: Figure out the labels stuff... If webhook does it, we can filter like this...
+	// Find all the Triggers that have been labeled as belonging to me
+	/*
+		triggers, err := r.triggerLister.Triggers(b.Namespace).List(labels.SelectorFromSet(brokerLabels(b.brokerClass)))
+	*/
+	triggers, err := r.triggerLister.Triggers(b.Namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	ctx = contextWithBroker(ctx, b)
+	ctx = contextWithTargets(ctx, r.targetsConfig)
+	for _, t := range triggers {
+		if t.Spec.Broker == b.Name {
+			logger := logging.FromContext(ctx).With(zap.String("trigger", t.Name), zap.String("broker", b.Name))
+			ctx = logging.WithLogger(ctx, logger)
+
+			if tKey, err := cache.MetaNamespaceKeyFunc(t); err == nil {
+				err = r.triggerReconciler.Reconcile(ctx, tKey)
+			}
+		}
+	}
+
+	//TODO aggregate errors?
+	return err
+}
+
+//TODO all this stuff should be in a configmap variant of the config object
+
+// This function is not thread-safe and should only be executed by
+// TargetsConfigUpdater
+func (r *Reconciler) updateTargetsConfig(ctx context.Context) error {
+	logger := r.Logger.Desugar()
+
+	//Load the existing config first if it exists
+	//TODO retry?
+	r.loadConfigOnce.Do(func() { r.loadTargetsConfig(ctx) })
+	//TODO resources package?
+	data, err := r.targetsConfig.Bytes()
+	if err != nil {
+		return fmt.Errorf("error serializing targets config: %w", err)
+	}
+	desired := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      targetsCMName,
+			Namespace: targetsCMNamespace,
+		},
+		BinaryData: map[string][]byte{targetsCMKey: data},
+		// For debugging purposes
```

Comment:
Should we just save the string instead of the binary format? This makes debug in the data plane easier too.

Reply:
I think I'd prefer the standard contract to be binary for efficiency and size limit reasons. This debugging one can be for humans only and change format or be removed if needed.

Reply:
Do we have an estimate of the actual size of the configmap (say for 50 brokers)? We should really try not to be bitten by any size limit any more